### PR TITLE
flow control using total writing bytes

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -55,9 +55,9 @@
 # also should less than total cpu cores.
 # scheduler-worker-pool-size = 4
 
-# When the number of concurrent running writes exceeds this threshold,
-# the "too busy" error is displayed.
-# scheduler-too-busy-threshold = 10000
+# When the pending write bytes exceeds this threshold,
+# the "scheduler too busy" error is displayed.
+# scheduler_pending_write_threshold = "10MB"
```
Comment (siddontang) [inline etc/config-template.toml]: s/scheduler_pending_write_threshold/scheduler-pending-write-threshold/

**Comment by siddontang** [comment]
LGTM

File: src/storage/config.rs [hunk 2]
```diff
@@ -24,10 +24,10 @@ const DEFAULT_SCHED_CAPACITY: usize = 10240;
...
 const DEFAULT_SCHED_CONCURRENCY: usize = 102400;
 
-// According to "Little's law", assuming you can write 100_000 KVs
-// per second, and it takes about 100ms to process the write requests
-// on average, hence using the 10_000 as the default value here.
-const DEFAULT_SCHED_TOO_BUSY_THRESHOLD: usize = 10_000;
+// According to "Little's law", assuming you can write 100MB per
+// second, and it takes about 100ms to process the write requests
+// on average, hence using the 10MB as the default value here.
+const DEFAULT_SCHED_PENDING_WRITE_MB: u64 = 10;
```
Comment (huachaohuang) [inline src/storage/config.rs]: I think we can tolerate a larger latency (e.g. 1s), so maybe 100MB is more proper?

**Pull request merged**